tensorboard/webapp/widgets/data_table/data_table_test.ts

```diff
@@ -456,4 +506,225 @@ describe('data table', () => {
+
+    beforeEach(() => {
+      mockHeaders = [
+        {
+          name: 'run',
+          type: ColumnHeaderType.RUN,
+          displayName: 'Run',
+          enabled: true,
+        },
+        {
+          name: 'disabled_header',
+          type: ColumnHeaderType.MAX_VALUE,
+          displayName: 'disabled',
+          enabled: false,
+        },
+        {
+          name: 'other_header',
+          type: ColumnHeaderType.HPARAM,
+          displayName: 'Display This',
+          enabled: true,
+        },
+      ];
+      mockTableData = [
+        {
+          id: 'runid',
+          run: 'run name',
+          disabled_header: 'disabled header',
+          other_header: 'other header',
+        },
+      ];
+      mockSelectableColumns = [
+        {
+          type: ColumnHeaderType.HPARAM,
+          name: 'lr',
+          displayName: 'Learning Rate',
+          enabled: false,
+        },
+      ];
+    });
+    it('renders context menu when a column cell is clicked', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      expect(fixture.debugElement.query(By.css('.context-menu'))).toBeNull();
+      const cell = fixture.debugElement.query(
+        By.directive(ContentCellComponent)
+      );
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      expect(
+        fixture.debugElement.query(By.css('.context-menu'))
+      ).not.toBeNull();
+    });
+
+    it('renders context menu when a column header is clicked', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      expect(fixture.debugElement.query(By.css('.context-menu'))).toBeNull();
+      const cell = fixture.debugElement.query(
+        By.directive(HeaderCellComponent)
+      );
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      expect(
+        fixture.debugElement.query(By.css('.context-menu'))
+      ).not.toBeNull();
+    });
+
+    it('renders column selector when add column to the left is clicked', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      const cell = fixture.debugElement.query(
+        By.directive(ContentCellComponent)
+      );
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      fixture.debugElement
+        .queryAll(By.css('.context-menu button'))
+        .find((element) => element.nativeElement.innerHTML.includes('Left'))!
+        .nativeElement.click();
+      fixture.detectChanges();
+
+      expect(
+        fixture.debugElement.query(By.directive(ColumnSelectorComponent))
+      ).not.toBeNull();
```

Comment:
Can you also check the contextMenuHeader and insertColumnTo are set correctly here and below?